[otbn] document/comment modexp otbn assembly lib

felixmiller opened the pull request:
Document modexp otbn assembly code

This finishes #2908

rswarbrick commented on sw/otbn/code-snippets/modexp.S:
```diff
@@ -203,80 +626,324 @@ setupPtrs:
   ecall
 
+
+/* Constant-time Mongomery Modular Multiplication
```
Nit: "Montgomery"

rswarbrick commented on sw/otbn/code-snippets/modexp.S:
```diff
@@ -55,38 +193,105 @@ computeRR:
+    /* reset flags of FG1 */
+    bn.sub    w3, w3, w3, FG1
+
+    /* compare intermediate biginit y with modulus
```
Nit: bigint

rswarbrick commented on sw/otbn/code-snippets/modexp.S:
```diff
@@ -319,41 +1046,164 @@ mulx_exp:
   ret
 
+
+/* Constant-time biginit modular exponentiation
```
Nit: biginit

rswarbrick reviewed (approved):
Wow, this is detailed! I really like all the comments, and feel like I understand things a bit better now, which is nice.

As well as the nitty typos I noticed, I think you've written "Clobbered Flag Groups" with capitals in the second half of the file and without in the first. (Yes, it's really pedantic, but I thought you'd like to know that someone actually read this :-D)

imphil reviewed (approved):
Great work Felix, thank you! 

On comments: You sometimes go for doxygen-style comments (`/**`), and sometimes for normal block comments (`/*`). Both is fine from my PoV, but it would be nice to make it consistent.

felixmiller commented:
Thanks @rswarbrick and @imphil! Pushed an update addressing your comments. If there is more please don't hesitate to point it out. This is a text-heavy commit, so if there is anything w.r.t. to terminology, style, use of English language, letter case, etc., happy to improve/correct.
